src/hocon.erl

```diff
@@ -400,11 +400,29 @@ do_concat(Concat) ->
 do_concat([], [{}]) ->
     {[]};
 do_concat([], [Object| _Objects] = Acc) when is_tuple(Object) ->
```

Comment:
how about 
```
-define(IS_OBJECT(O), (is_tuple(O) andalso size(O) =:= 1)).
```

and below to replace `{[...]}`.
```
-define(OBJECT(FIELDS), {FIELDS}).
```

Reply:
```
-define(IS_OBJECT(O), (is_tuple(O) andalso size(O) =:= 1 andalso is_list(element(1, O)))).
-define(IS_FIELD(F), (is_tuple(F) andalso size(F) =:= 2)).
-define(OBJECT(FIELDS), {FIELDS}).
-define(FIELDS(OBJECT), element(1, OBJECT)).
```